### DISCUSSION TITLE
Combining Automatic1111 & Oobabooga?

### DISCUSSION THREAD

**Discussion by gitihobo** [oobabooga/text-generation-webui#1481] [Ideas]
So send pictures is the start of the collaboration, but since both projects have the same goals, wouldn't it be good to turn it into a single repo? There would be more innovation because we can focus on making both Tecnologies better combining them.

**Comment by ThioJoe** [comment]
Oh god, it would become hideously complex and bloated

**Reply by gitihobo** [reply]
It could, we would have to become way more efficient with organization, completely overhaul the GUIs by starting simple and allowing people to add complexity through settings but I think overall it would evolve the technology for the better.

**Comment by kuso-ge** [comment]
Nah I like that its separate. It allows the project contributors to focus on tasks at hand or specific goals. While the technology might have similarities in some aspects but there are distinct differences and that requires a more focused development.

**Reply by gitihobo** [reply]
That is true, what about an official flexible fork? Could that work?

**Reply by kuso-ge** [reply]
maybe if someone is willing to try. I assume it's gonna be hard to maintain since you don't really know which directions either repo will take (since they have different focus and goals) and it may easily break. 
Also on the hardware side, although I can run both Ooba and Auto's webui side by side, I have to run Ooba and load the 4-bit models first since they require a huge amount of RAM to preload the model (15GB~ gets filled) before being passed to the GPU. (I have a 3060 12GB GPU, 16GB RAM). With how much resource intensive both of these are, I assume most people won't be able to run it with low-mid specs and might require more high end systems just for both to run simultaneously.

**Reply by gitihobo** [reply]
This is all quite valid, in terms of things breaking I assume it takes time but is fixable in the end of the day, It may be too early to think about combining but I also think specs will lower with time and the potential upside is great, it's already half starting with things like ooba SD extension and minichatGPT being able to analyze images in chat. How is ooba or 1111 currently monetizing? Is there a main team?

**Comment by AIZeplin** [comment]
I would much prefer extensions that could be added to each. Combining them into a singular project seems crazy complex - SD is already madly complex to learn, would be wild with this on top. I think extensions between them would be better, at least until the tech of each is more developed.

**Reply by gitihobo** [reply]
Yeah I think that's the popular opinion and honestly I think extensions are currently the way. Still I think the quicker we combine this technology in fun ways with other tech the better
